### ISSUE TITLE
Add codecov integration

### ISSUE THREAD

**Issue by jwillemsen** [objectcomputing/OpenDDS#593]
Would be nice to also have codecov support so that we have integrated code coverage data

**Comment by mitza-oci** [comment]
Is it set up for ACE_TAO?

**Comment by jwillemsen** [comment]
No, it has not been setup for ACE/TAO, we have no coverage build at this moment. Because there is an OpenDDS coverage build it is maybe easier to add it there, according to their documentation it is just one command to be added, maybe something just to try, when it works it would be nice, if not, it can be put on the backlog

**Comment by jwillemsen** [comment]
Following will maybe work when added to the OpenDDS code coverage build, where foo has to be the real token
```
<command name="anonymous_shell" options="curl -s https://codecov.io/bash -o $DOC_ROOT/cov.sh" />
<command name="anonymous_shell" options="bash $DOC_ROOT/cov.sh -t foo -R objectcomputing/OpenDDS -s $DDS_ROOT" />
```

**Comment by jwillemsen** [comment]
Made some test uploads with only ACE tests enabled, see https://codecov.io/gh/DOCGroup/ACE_TAO and https://coveralls.io/github/DOCGroup/ACE_TAO

**Comment by jwillemsen** [comment]
Now that github actions are used, this could be done using the @codecov/codecov-action@v1@ action, see https://github.com/RemedyIT/taox11/pull/150

**Comment by jwillemsen** [comment]
We are using codecov now with taox11 and it works better as I expected. We build the code with gcov enabled and run the tests as part of the github CI action and run the codecov action which gathers all code coverage and updates the PR, see for example https://github.com/RemedyIT/taox11/pull/152

**Comment by jwillemsen** [comment]
See https://github.com/RemedyIT/taox11/blob/0b7a6a0a4bb58181c554501bd6f0b4dd008f8a13/.github/workflows/linux.yml#L144 for the action to add codecov
